### DISCUSSION TITLE
Is NVIDA GPU Hotplug supported now ?

### DISCUSSION THREAD

**Discussion by liuyd96** [NVIDIA/open-gpu-kernel-modules#451] [Q&A] [answered]
Hi, Guys.
I want to know if NVDIA GPU Hot(un)plug is supported now.
I noticed src/nvidia-modeset/src/nvkms-event.c has mentioned **Hotplug**
```
 35 /*
 36  * Handle a display device hotplug event.
 37  *
 38  * What "hotplug" means is unclear, but it could mean any of the following:
 39  *   - A display device is plugged in.
 40  *   - A display device is unlugged.
 41  *   - A display device was unplugged and then plugged back in.
 42  *   - A display device was plugged in and then unplugged.
 43  *   - An already connected display device is turned on.
 44  *   - An already connected display device is turned off.
 45  *   - A DisplayPort device needs its link status and RX Capabilities fields
 46  *     read and may need to be retrained ("long" hotplug event, > 2ms).
 47  *
 48  * DisplayPort "short" hotplug events, which are between 0.25ms and 2ms, are
 49  * handled separately by nvHandleDPIRQEventDeferredWork below.
 50  */
 51  
```
I’ve tried to remove GPU in linux by “echo 1 > /sys/bys/pci/devices/0000:****/remove” and that works. 
After that, could I unplug the GPU directly?

**Comment by aritger** [comment, marked as answer]
The comments you quote from nvkms-event.c are referring to monitor hotplug, not GPU hotplug.

 I suspect there is a lot of work still necessary to reliably support GPU hotplug/hotunplug.

**Reply by liuyd96** [reply]
Thank you very much for your answer.

**Reply by Bruce1998** [reply]
Nvidia may be  able to support hot plug now, I noticed that there are already hot plug graphics card docking stations on the market, and they indicate that  only support Nvidia
